pkg/exasol-rest-api/integration_test.go

```diff
@@ -0,0 +1,207 @@
+package exasol_rest_api_test
+
+import (
+	"context"
+	"database/sql"
+	"github.com/gin-gonic/gin"
+	"log"
+	exasol_rest_api "main/pkg/exasol-rest-api"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/exasol/exasol-driver-go"
+	"github.com/stretchr/testify/suite"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+	ctx                   context.Context
+	exasolContainer       testcontainers.Container
+	defaultExasolUsername string
+	defaultExasolPassword string
+	exasolPort            int
+	appProperties         *exasol_rest_api.ApplicationProperties
+}
+
+func TestIntegrationSuite(t *testing.T) {
+	suite.Run(t, new(IntegrationTestSuite))
+}
+
+func (suite *IntegrationTestSuite) SetupSuite() {
+	suite.ctx = context.Background()
+	suite.defaultExasolUsername = "api_service_account"
+	suite.defaultExasolPassword = "secret_password"
+	suite.exasolContainer = runExasolContainer(suite.ctx)
+	suite.exasolPort = getExasolPort(suite.exasolContainer, suite.ctx)
+	createDefaultServiceUserWithAccess(suite.defaultExasolUsername, suite.defaultExasolPassword, suite.exasolPort)
+}
+
+func (suite *IntegrationTestSuite) startServer(application exasol_rest_api.Application) *gin.Engine {
+	gin.SetMode(gin.TestMode)
+	router := gin.Default()
+	router.GET("/api/v1/query/:query", application.Query)
+	suite.appProperties = application.Properties
+	return router
+}
+
+func (suite *IntegrationTestSuite) TestQuery() {
+	router := suite.startServer(suite.createApplicationWithDefaultProperties())
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/SELECT * FROM TEST_SCHEMA_1.TEST_TABLE", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusOK, responseRecorder.Code)
+	suite.Equal("{\"status\":\"ok\",\"responseData\":{\"results\":[{\"resultType\":\"resultSet\",\"resultSet\":{\"numColumns\":2,\"numRows\":1,\"numRowsInMessage\":1,\"columns\":[{\"name\":\"X\",\"dataType\":{\"type\":\"DECIMAL\",\"precision\":18,\"scale\":0}},{\"name\":\"Y\",\"dataType\":{\"type\":\"VARCHAR\",\"size\":100,\"characterSet\":\"UTF8\"}}],\"data\":[[15],[\"test\"]]}}],\"numResults\":1}}",
+		responseRecorder.Body.String())
+}
+
+func (suite *IntegrationTestSuite) TestInsertNotAllowed() {
+	router := suite.startServer(suite.createApplicationWithDefaultProperties())
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/CREATE SCHEMA not_allowed_schema", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusOK, responseRecorder.Code)
+	suite.Contains(responseRecorder.Body.String(),
```

Comment:
In java, we do `assertAll`. No idea if there is something similar for go.
By the way, there is also hamcrest for go. Maybe we should use it.

Reply:
I don't want to switch the testing framework right now and this one doesn't have assertAll. I'll add an idea about assertAll to this ticket: https://github.com/exasol/exasol-rest-api/issues/13